File: client/web/sudoo/lib/domain/model/order/order.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:sudoo/domain/model/payment/payment.dart';
import 'package:sudoo/domain/model/promotion/promotion_info.dart';
import 'package:sudoo/domain/model/user/user.dart';

import 'order_supplier.dart';

part 'order.g.dart';

@JsonSerializable(explicitToJson: true)
class Order {
  final String orderId;
  final String cartId;
  final Payment payment;
  final PromotionInfo? promotion;
  final User user;
  final String address;
  final double totalPrice;
  final double totalPromotionPrice;
  final double finalPrice;
  final double totalShipmentPrice;
  final DateTime createdAt;
  final List<OrderSupplier> orderSuppliers;

  Order(
    this.orderId,
    this.cartId,
    this.payment,
    this.promotion,
    this.user,
    this.address,
    this.totalPrice,
    this.totalPromotionPrice,
    this.finalPrice,
    this.totalShipmentPrice,
    this.createdAt,
    this.orderSuppliers,
  );

  factory Order.fromJson(Map<String, dynamic> json) => _$OrderFromJson(json);

  Map<String, dynamic> toJson() => _$OrderToJson(this);
}

File: client/mobile/ShipperDemo/app/src/main/java/com/sudoo/shipment/model/Order.kt
package com.sudoo.shipment.model

import java.time.LocalDateTime

data class Order(
    val orderId: String,
    val cartId: String,
    val payment: PaymentInfo? = null,
    val promotion: PromotionInfo? = null,
    val user: User,
    val totalPrice: Double,
    val totalPromotionPrice: Double,
    val totalShipmentPrice: Double,
    val finalPrice: Double,
    val address: String,
    val createdAt: LocalDateTime,
    val orderSuppliers: List<OrderSupplier>
)
